contents/blog/clickhouse-vs-snowflake.md

```diff
@@ -21,105 +21,111 @@ If I was to compare the databases to boats, ClickHouse is a jet ski – ultra-fa
 
-- Snowflake's [Variant type](https://docs.snowflake.com/en/sql-reference/data-types-semistructured) was built into the service from the beginning and makes querying unstructured data nearly effortless. it basically decomposes JSON objects into a table of key-value pairs that better leverages the columnar aspect of Snowflake's architecture. This is something ClickHouse is actively working on, but for now is a competitive advantage of Snowflake.
+Snowflake has some of these features, but with caveats: 
 
-Overall, ClickHouse's close-metal optimizations enable it to be faster than Snowflake and have more flexibility in how you can deploy it. Snowflake is, and forever will be, cloud only. ClickHouse can be deployed on-prem, in the cloud, or even on your laptop. You can even run it on a Raspberry Pi! The best bit about ClickHouse is that you can progressively tier your data from being local to block storage so that you have the best of all worlds. 
+- While Snowflake has support for materialized views, it charges additionally for it since it needs to allocate serious CPU resources to recompute the views. Those re-computations, under the hood, are more resource intensive for Snowflake because it lacks the same partial-update techniques ClickHouse uses to expedite re-calculations.
 
-### Who uses Snowflake and ClickHouse?
+- **Virtual warehouses instead of specialized engines:** Unlike ClickHouse's specialized engines, which utilize hardware to optimize query execution, Snowflake segments its database into virtual warehouses sized from small to large. These warehouses can have larger or lesser compute to handle complex to easy queries and isolate workloads. However, while clever and organized, this approach is computationally and financially expensive when taken full advantage of.
 
-Part of Snowflake’s value prop is the broad appeal to both technical and non-technical users. Snowflake’s marketplace makes connecting business intelligence tools with warehouse data easy, which appeals to teams wanting explore their data. ClickHouse Cloud has a competitive offering and the ecosystem is growing fast, but still smaller than Snowflake’s.
+- **Search optimization service:** Introduced in 2021, Snowflake offers a search optimization service at its higher enterprise tiers. Because Snowflake is a closed-source product, not much is known about how the search optimization service works, but it can offer enormous performance gains (4x-100x) relative to un-optimized search for some types of queries. Unfortunately, while Snowflake likely uses similar mechanisms to ClickHouse, direct side-by-side analysis isn’t possible due to Snowflake’s closed-search tier. By extension, the search optimization service’s performance boost is unpredictable.
```

Comment:
Unless there's something factually incorrect in these sections, I'm inclined to put them back in.